crates/map/src/finders/dfs.cairo

```diff
@@ -0,0 +1,184 @@
+//! Depth-First Search algorithm implementation for pathfinding.
+
+// Core imports
+use core::dict::{Felt252Dict, Felt252DictTrait};
+
+// Internal imports
+use origami_map::finders::finder::Finder;
+use origami_map::helpers::bitmap::Bitmap;
+use origami_map::helpers::seeder::Seeder;
+use origami_map::types::node::{Node, NodeTrait};
+use origami_map::types::direction::{Direction, DirectionTrait};
+
+/// DepthFirstSearch implementation for pathfinding
+#[generate_trait]
+pub impl DFS of DFSTrait {
+    /// Searches for a path from 'from' to 'to' on the given grid using DepthFirstSearch
+    ///
+    /// # Arguments
+    /// * `grid` - The grid represented as a felt252
+    /// * `width` - The width of the grid
+    /// * `height` - The height of the grid
+    /// * `from` - The starting position
+    /// * `to` - The target position
+    ///
+    /// # Returns
+    /// A Span<u8> representing the path from 'from' to 'to', or an empty span if no path exists
+    #[inline]
+    fn search(grid: felt252, width: u8, height: u8, from: u8, to: u8) -> Span<u8> {
+        // [Check] The start and target are walkable
+        if Bitmap::get(grid, from) == 0 || Bitmap::get(grid, to) == 0 {
+            return array![].span();
+        }
+
+        // [Effect] Initialize the start and target nodes
+        let start = NodeTrait::new(from, 0, 0, 0);
+        let target = NodeTrait::new(to, 0, 0, 0);
+
+        // [Effect] Initialize visited nodes and parents
+        let mut visited: Felt252Dict<bool> = Default::default();
+        let mut parents: Felt252Dict<u8> = Default::default();
+
+        // [Compute] Start the recursive DFS
+        let found = Self::dfs_recursive(
+            grid, width, height, start, target, ref visited, ref parents
+        );
+
+        // Reconstruct and return the path if found
+        if found {
+            Finder::path_with_parents(ref parents, start, target)
+        } else {
+            array![].span()
+        }
+    }
+
+    /// Recursive helper function for DFS
+    #[inline]
+    fn dfs_recursive(
```

Comment:
`iter` to be consistent with other recursions function in the crate: https://github.com/dojoengine/origami/blob/07f2b262b2653ba119f8e55eefe68cca18f4dc89/crates/map/src/generators/mazer.cairo#L52